demos/minimap/minimap.js

```diff
@@ -0,0 +1,247 @@
+/**
+* Blockly Demos: Code
+*
+* Copyright 2017 Google Inc.
+* https://developers.google.com/blockly/
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/**
+* @fileoverview JavaScript for Blockly's Minimap demo.
+* @author karnpurohit@gmail.com (Karan Purohit)
+*/
+'use strict';
+
+/**
+* Creating a seperate namespace for minimap.
+*/
+var Minimap = {};
+
+/**
+* initilize the workspace and minimap.
+* @param {Workspace} workspace The main workspace of the user
+* @param {Workspace} minimap The workspace that will be used as a minimap
+*/
+Minimap.init = function(workspace,minimap){
+  this.workspace= workspace;
+  this.minimap= minimap;
+
+  //Adding scroll callback functionlity to vScroll just for this demo.
+  //IMPORTANT: This should be changed when there is proper UI event handling
+  //           api available and should be handled by workspace's event listeners
+  this.workspace.scrollbar.vScroll.onScroll_ = function(){
+    var ratio = this.handlePosition_ / this.scrollViewSize_;
+    if (isNaN(ratio)) {
+      ratio = 0;
+    }
+    var xyRatio = {};
+    if (this.horizontal_) {
+      xyRatio.x = ratio;
+    } else {
+      xyRatio.y = ratio;
+    }
+    this.workspace_.setMetrics(xyRatio);
+
+    // get the absolutePosition
```

Comment:
Add a comment that the above code is what it normally does on scroll, and the parts below are yours.